task/git-clone-oci-ta/0.1/git_clone_oci_ta_test.go

```diff
@@ -0,0 +1,86 @@
+package git_clone_oci_ta_v01
+
+import (
+	"context"
+	_ "embed"
+	"fmt"
+	"regexp"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+	tekton "github.com/tektoncd/pipeline/pkg/apis/pipeline/v1"
+	meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/rand"
+
+	"github.com/konflux-ci/build-definitions/internal/config"
+	"github.com/konflux-ci/build-definitions/internal/kube"
+	"github.com/konflux-ci/build-definitions/internal/tkn"
+)
+
+//go:embed git-clone-oci-ta.yaml
+var taskDefinition []byte
+
+var scheme = runtime.NewScheme()
+
+func TestSimpleTest(t *testing.T) {
+	ctx := context.Background()
+
+	gitURL := "https://github.com/redhat-appstudio-qe/retrodep"
+	gitRevision := "main"
+	ociRepo := config.OCIRepo()
+	gitOCIStorage := fmt.Sprintf("%s:git-%s", ociRepo, rand.String(5))
+
+	taskSpec, err := tkn.TaskSpecFromDefinition(taskDefinition)
+	require.NoError(t, err)
+
+	taskRun := tekton.TaskRun{
+		ObjectMeta: meta.ObjectMeta{
+			GenerateName: "git-clone-oci-ta-v01-",
+			Namespace:    config.KubeNamespace(),
+		},
+		Spec: tekton.TaskRunSpec{
+			Params: []tekton.Param{
+				{Name: "url", Value: *tekton.NewStructuredValues(gitURL)},
+				{Name: "revision", Value: *tekton.NewStructuredValues(gitRevision)},
+				{Name: "ociStorage", Value: *tekton.NewStructuredValues(gitOCIStorage)},
+			},
+			TaskSpec: taskSpec,
+		},
+	}
+
+	k, err := kube.NewClient()
+	require.NoError(t, err)
+
+	require.NoError(t, k.Create(ctx, &taskRun))
+	defer k.Delete(ctx, &taskRun)
```

Comment:
What does it look like when a test fails? One of the big problems with our e2e-tests is that most of the time, the error message is pretty much useless on its own. If we don't leave the TaskRun around to debug, dealing with failing tests will be hard

Maybe the tests could capture the TaskRun yaml, logs, events and output/save them somewhere for debugging?

Reply:
Yeah. This is probably something that needs further refinement. My hope is that this is significantly easier to run than the current e2e tests. In the case of a failure, it should be very simple to re-run the test to reproduce the issue, and allow one to attach a debugger which is much better than whatever data is printed out.

I would like to try this as is and see if dumping all the additional information is really needed.

Reply:
For such kind of tests, it should be fine if saving TaskRun logs only. PR authors should be responsible for the correction of the proposed changes. From my point of view, a tests pass in the CI is just one of the acceptance criteria for approval. So, if there is any error causing the failure, PR authors can always re-run the test by themselves and they will have everything for debug.